Update warehouse classes

Criss-Wang opened the pull request:
A few things to note:
- ParserUtilTest still have some test cases not written for parseProduct
- the handling of warehouses in each Command is not written (command + test + parser + parser test)
- The Ui (fxml files) for warehouse is not ready
- The LogicManagerTest will need the correct command format for

Criss-Wang committed:
Reconstruct the file structure and add warehouse-related classes

Criss-Wang committed:
Update warehouse-related utility and test classes

The logic of the app needs to include warehouses.
The addressbook needs to store the warehouse cases.

Let's make the following changes:
- add getFilteredWarehouseList() and corresponding test.
- TODO: fix the execute_storageThrowsIoException test.
- TODO: include warehouse in command classes.

Criss-Wang committed:
More test case modifications

Criss-Wang committed:
Update remaining classes with warehouse integration

Criss-Wang committed:
Merge remote-tracking branch 'upstream/master' into create-warehouse-classes

# Conflicts:
#	docs/tutorials/RemovingFields.md
#	src/main/java/seedu/address/model/supplier/Address.java
#	src/main/java/seedu/address/model/supplier/Email.java
#	src/main/java/seedu/address/model/supplier/Name.java
#	src/main/java/seedu/address/model/supplier/NameContainsKeywordsPredicate.java
#	src/main/java/seedu/address/model/supplier/Phone.java
#	src/main/java/seedu/address/model/tag/Tag.java
#	src/main/java/seedu/address/model/warehouse/Remark.java
#	src/main/java/seedu/clinic/logic/Logic.java
#	src/main/java/seedu/clinic/logic/LogicManager.java
#	src/main/java/seedu/clinic/logic/commands/EditCommand.java
#	src/main/java/seedu/clinic/logic/commands/FindCommand.java
#	src/main/java/seedu/clinic/logic/parser/AddCommandParser.java
#	src/main/java/seedu/clinic/logic/parser/EditCommandParser.java
#	src/main/java/seedu/clinic/logic/parser/FindCommandParser.java
#	src/main/java/seedu/clinic/logic/parser/ParserUtil.java
#	src/main/java/seedu/clinic/model/Clinic.java
#	src/main/java/seedu/clinic/model/Model.java
#	src/main/java/seedu/clinic/model/ModelManager.java
#	src/main/java/seedu/clinic/model/ReadOnlyClinic.java
#	src/main/java/seedu/clinic/model/attribute/Address.java
#	src/main/java/seedu/clinic/model/attribute/Email.java
#	src/main/java/seedu/clinic/model/attribute/Name.java
#	src/main/java/seedu/clinic/model/attribute/NameContainsKeywordsPredicateForSupplier.java
#	src/main/java/seedu/clinic/model/attribute/NameContainsKeywordsPredicateForWarehouse.java
#	src/main/java/seedu/clinic/model/attribute/Phone.java
#	src/main/java/seedu/clinic/model/attribute/Remark.java
#	src/main/java/seedu/clinic/model/attribute/Tag.java
#	src/main/java/seedu/clinic/model/product/Product.java
#	src/main/java/seedu/clinic/model/supplier/Address.java
#	src/main/java/seedu/clinic/model/supplier/Email.java
#	src/main/java/seedu/clinic/model/supplier/Name.java
#	src/main/java/seedu/clinic/model/supplier/NameContainsKeywordsPredicate.java
#	src/main/java/seedu/clinic/model/supplier/Phone.java
#	src/main/java/seedu/clinic/model/supplier/Remark.java
#	src/main/java/seedu/clinic/model/supplier/Supplier.java
#	src/main/java/seedu/clinic/model/tag/Tag.java
#	src/main/java/seedu/clinic/model/util/SampleDataUtil.java
#	src/main/java/seedu/clinic/model/warehouse/Remark.java
#	src/main/java/seedu/clinic/model/warehouse/UniqueWarehouseList.java
#	src/main/java/seedu/clinic/model/warehouse/Warehouse.java
#	src/main/java/seedu/clinic/storage/JsonAdaptedProduct.java
#	src/main/java/seedu/clinic/storage/JsonAdaptedSupplier.java
#	src/main/java/seedu/clinic/storage/JsonAdaptedTag.java
#	src/main/java/seedu/clinic/storage/JsonAdaptedWarehouse.java
#	src/main/java/seedu/clinic/storage/JsonSerializableClinic.java
#	src/main/java/seedu/clinic/ui/WarehouseCard.java
#	src/main/java/seedu/clinic/ui/WarehouseListPanel.java
#	src/test/data/JsonClinicStorageTest/invalidAndValidWarehouseAddressBook.json
#	src/test/data/JsonClinicStorageTest/invalidWarehouseAddressBook.json
#	src/test/data/JsonSerializableClinicTest/duplicateWarehouseAddressBook.json
#	src/test/data/JsonSerializableClinicTest/invalidWarehouseAddressBook.json
#	src/test/data/JsonSerializableClinicTest/typicalWarehousesAddressBook.json
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/supplier/AddressTest.java
#	src/test/java/seedu/address/model/supplier/EmailTest.java
#	src/test/java/seedu/address/model/supplier/NameContainsKeywordsPredicateTest.java
#	src/test/java/seedu/address/model/supplier/NameTest.java
#	src/test/java/seedu/address/model/supplier/PhoneTest.java
#	src/test/java/seedu/address/model/tag/TagTest.java
#	src/test/java/seedu/address/model/warehouse/RemarkTest.java
#	src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java
#	src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java
#	src/test/java/seedu/address/testutil/AddressBookBuilder.java
#	src/test/java/seedu/clinic/logic/LogicManagerTest.java
#	src/test/java/seedu/clinic/logic/commands/CommandTestUtil.java
#	src/test/java/seedu/clinic/logic/commands/EditWarehouseDescriptorTest.java
#	src/test/java/seedu/clinic/logic/commands/FindCommandTest.java
#	src/test/java/seedu/clinic/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/clinic/logic/parser/ClinicParserTest.java
#	src/test/java/seedu/clinic/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/clinic/logic/parser/FindCommandParserTest.java
#	src/test/java/seedu/clinic/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/clinic/model/ModelManagerTest.java
#	src/test/java/seedu/clinic/model/attribute/AddressTest.java
#	src/test/java/seedu/clinic/model/attribute/EmailTest.java
#	src/test/java/seedu/clinic/model/attribute/NameContainsKeywordsPredicateForSupplierTest.java
#	src/test/java/seedu/clinic/model/attribute/NameContainsKeywordsPredicateForWarehouseTest.java
#	src/test/java/seedu/clinic/model/attribute/NameTest.java
#	src/test/java/seedu/clinic/model/attribute/PhoneTest.java
#	src/test/java/seedu/clinic/model/attribute/RemarkTest.java
#	src/test/java/seedu/clinic/model/attribute/TagTest.java
#	src/test/java/seedu/clinic/model/supplier/AddressTest.java
#	src/test/java/seedu/clinic/model/supplier/EmailTest.java
#	src/test/java/seedu/clinic/model/supplier/NameContainsKeywordsPredicateTest.java
#	src/test/java/seedu/clinic/model/supplier/NameTest.java
#	src/test/java/seedu/clinic/model/supplier/PhoneTest.java
#	src/test/java/seedu/clinic/model/tag/TagTest.java
#	src/test/java/seedu/clinic/model/warehouse/RemarkTest.java
#	src/test/java/seedu/clinic/model/warehouse/UniqueWarehouseListTest.java
#	src/test/java/seedu/clinic/model/warehouse/WarehouseTest.java
#	src/test/java/seedu/clinic/storage/JsonAdaptedSupplierTest.java
#	src/test/java/seedu/clinic/storage/JsonAdaptedWarehouseTest.java
#	src/test/java/seedu/clinic/testutil/EditSupplierDescriptorBuilder.java
#	src/test/java/seedu/clinic/testutil/EditWarehouseDescriptiorBuilder.java
#	src/test/java/seedu/clinic/testutil/SupplierBuilder.java
#	src/test/java/seedu/clinic/testutil/TestUtil.java
#	src/test/java/seedu/clinic/testutil/TypicalSupplier.java
#	src/test/java/seedu/clinic/testutil/TypicalWarehouse.java
#	src/test/java/seedu/clinic/testutil/WarehouseBuilder.java

Criss-Wang committed:
Merge with refactored package naming and clean up code

Criss-Wang committed:
Fix minor issues with refactoring

Criss-Wang committed:
Merge branch 'master' into create-warehouse-classes

Criss-Wang committed:
Fix EOF styling issue

Criss-Wang committed:
Revert change in MainWindow to restore WarehouseListPanel back

tohyuting commented on src/main/java/seedu/clinic/model/Clinic.java:
```diff
@@ -3,18 +3,22 @@
 /**
  * Wraps all data at the clinic level
- * Duplicates are not allowed (by .isSameSupplier comparison)
+ * Duplicates are not allowed (by .isSameSupplier and isSameWarehouse comparison)
```
Minor issue, missing '.' for isSameWarehouse

tohyuting commented on src/main/java/seedu/clinic/model/attribute/Tag.java:
```diff
@@ -1,10 +1,10 @@
 
 /**
- * Represents a Tag in the CLI-nic app.
+ * Represents a tag for the product related to a specific supplier/warehouse in CLI-nic app.
```
Should we include warehouse here although for this iteration we are not intending to include tagging for warehouses products?

tohyuting commented on src/main/java/seedu/clinic/model/supplier/UniqueSupplierList.java:
```diff
@@ -127,9 +127,7 @@ public int hashCode() {
-                if (suppliers.get(i).isSameSupplier(suppliers.get(j))) {
-                    return false;
-                }
+                return !suppliers.get(i).isSameSupplier(suppliers.get(j));
```
With the if condition statement removed here, will there be an issue where the for loop only runs one time since the return statement will force the function to stop executing?

tohyuting commented on src/main/java/seedu/clinic/model/warehouse/UniqueWarehouseList.java:
```diff
@@ -132,9 +130,7 @@ public int hashCode() {
-                if (warehouses.get(i).isSameWarehouse(warehouses.get(j))) {
-                    return false;
-                }
+                return !warehouses.get(i).isSameWarehouse(warehouses.get(j));
```
Same issue here, as pointed out above.

tohyuting commented on src/main/resources/view/MainWindow.fxml:
```diff
@@ -60,6 +60,22 @@
+<!--          <StackPane fx:id="warehouseListPanelPlaceholder" VBox.vgrow="ALWAYS"/>-->
+<!--        </VBox>-->
+
+        <StackPane fx:id="statusbarPlaceholder" VBox.vgrow="NEVER" />
```
Is the extra supplierList here as a placeholder for warehouse? I think the statusbarPlaceholder should be removed as well :)

tohyuting commented on src/test/data/JsonSerializableClinicTest/duplicateWarehouseClinic.json:
```diff
@@ -0,0 +1,22 @@
+      "name": "Panadol",
+      "quantity": 20
+    }]
+  } ]
```
Just adding a note here to remember to change supplier to warehouse when updating tests cases later on.

tohyuting commented on src/test/java/seedu/clinic/logic/parser/AddCommandParserTest.java:
```diff
@@ -28,13 +28,23 @@
 
 import org.junit.jupiter.api.Test;
 
+<<<<<<< HEAD:src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.model.attribute.Email;
+import seedu.address.model.attribute.Name;
+import seedu.address.model.attribute.Phone;
+import seedu.address.model.attribute.Remark;
+import seedu.address.model.supplier.Supplier;
+import seedu.address.testutil.SupplierBuilder;
+=======
 import seedu.clinic.logic.commands.AddCommand;
 import seedu.clinic.model.supplier.Email;
 import seedu.clinic.model.supplier.Name;
 import seedu.clinic.model.supplier.Phone;
 import seedu.clinic.model.supplier.Remark;
 import seedu.clinic.model.supplier.Supplier;
 import seedu.clinic.testutil.SupplierBuilder;
+>>>>>>> upstream/master:src/test/java/seedu/clinic/logic/parser/AddCommandParserTest.java
```
Just to check, did you perhaps miss out the conflict here?

tohyuting commented on src/test/java/seedu/clinic/logic/parser/EditCommandParserTest.java:
```diff
@@ -42,6 +53,7 @@
 import seedu.clinic.model.supplier.Phone;
 import seedu.clinic.model.tag.Tag;
 import seedu.clinic.testutil.EditSupplierDescriptorBuilder;
+>>>>>>> upstream/master:src/test/java/seedu/clinic/logic/parser/EditCommandParserTest.java
```
Same for these conflicted import statements as well :)

tohyuting reviewed (commented):
Just some minor issues, otherwise great job!

Criss-Wang committed:
Fix logic issue in uniqueness test and MainWindow component display

Criss-Wang committed:
fix checkstyle issue

Criss-Wang committed:
fix checkstyle issue

zhengweii commented on docs/UserGuide.md:
```diff
@@ -13,7 +13,7 @@ and efficient Graphical User Interface interaction.
 --------------------------------------------------------------------------------------------------------------------
 
-## Quick start
+## Quick startf
```
Oops minor typo here!

zhengweii commented on src/main/java/seedu/clinic/model/attribute/Name.java:
```diff
@@ -13,10 +13,10 @@
-     * The first character of the name must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * The first character of the name must be alphanumeric.
+     * The .
```
I think this should be removed as well!

Criss-Wang committed:
Fix display issue for warehouse.

The warehouse list was not displayed due the a minor bug in the code.
An error was thrown suggesting invalid escape sequence. We need to find
the attribute which has a type and fix it.

Let's fix the bug found in warehouseListCard.fxml.

tohyuting reviewed (approved):
LGTM
